File: KersDb/Entities/KERScore/Alert.cs
using System;
using Kers.Models.Entities.KERSmain;
using Kers.Models.Abstract;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Kers.Models.Entities.KERScore
{
    public partial class Alert:IEntityBase, IEntityCredentials
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id {get; set;}
        public string Message {get; set;}
        public string UrlRoute {get; set;}
        public int AlertType {get;set;}
        public DateTime Start {get;set;}
        public DateTime End {get;set;}
        public string MoreInfoUrl {get;set;}
        public int? EmployeePositionId {get; set;}
        public int? zEmpRoleTypeId {get; set;}
        public int? isContyStaff {get;set;}
        public ExtensionRegion ExtensionRegion {get;set;}
        public int? ExtensionRegionId {get;set;}
        public ExtensionArea ExtensionArea {get;set;}
        public int? ExtensionAreaId {get;set;}
        public PlanningUnit PlanningUnit {get;set;}
        public int? PlanningUnitId {get;set;}
        public bool Active {get;set;}
        public KersUser CreatedBy {get;set;}
        public DateTime Created {get;set;}
        public DateTime LastUpdated {get;set;}
        
    }
}

File: Kers/ClientApp/src/app/modules/reporting/modules/alerts/Alert.ts
import { KersUser } from "../admin/users/users.service";

export class Alert{
    id:number;
    message:string;
    urlRoute:string;
    alertType:number;
    start:Date;
    end:Date;
    moreInfoUrl:string;
    active:boolean;
    createdBy:KersUser;
    created:Date;
    lastUpdated:Date;
    employeePositionId?:number;
    zEmpRoleTypeId?:number;
    isCountyStarr?:number;
}

export class AlertRoute{
    id:number;
    name:string;
    urlRoute:string;
    active:boolean;
}

export const AlertTypes:AlertType[] =   [
    {
        id:1,name:"Note (green)",class:"alert-success"
    },
    {
        id:2,name:"Warning (blue)",class:"alert-info"
    },
    {
        id:3,name:"Alert (orange)",class:"alert-warning"
    },
    {
        id:4,name:"Urgent (red)",class:"alert-danger"
    }
                                        ];


export class AlertType{
    id:number;
    name: string;
    class: string;
}